cartoframes/data/dataset.py

```diff
@@ -25,20 +22,15 @@ class Dataset(object):
     GEOM_TYPE_POLYGON = GEOM_TYPE_POLYGON
 
     def __init__(self, data, credentials=None, schema=None):
+        self._registry = self._get_strategies_registry()
         self._strategy = self._init_strategy(data, credentials, schema)
         self._is_saved_in_carto = self._init_saved_in_carto()
 
     def _init_strategy(self, data, credentials=None, schema=None):
         credentials = credentials or _get_default_credentials()
-
-        if isinstance(data, pd.DataFrame):
-            return self._getDataFrameDataset(data)
-        elif is_geojson(data):
-            return self._getDataFrameDataset(load_geojson(data))
-        elif is_sql_query(data):
-            return self._getQueryDataset(data, credentials)
-        elif is_table_name(data):
-            return self._getTableDataset(data, credentials, schema)
+        for strategy in self._registry.get_strategies():
+            if strategy.can_work_with(data):
+                return strategy.create(data, credentials, schema)
 
         raise ValueError('We can not detect the Dataset type for data={}'.format(data))
```

Comment:
Since data can be an arbitrary object, we might want not to print it.

Reply:
Yep, but it is good to know what happens... :thinking:  Maybe for the users, it is not very relevant. Oj I will remove it